lua/gitlinker/routers.lua

```diff
@@ -75,6 +75,47 @@ function Builder:new(lk, range_maker)
   return o
 end
 
+--- @param s string
+--- @param t string
+local function string_endswith(s, t)
+  return string.len(s) >= string.len(t) and string.sub(s, #s - #t + 1) == t
+end
+
+--- @param lk gitlinker.Linker
+--- @return string
+local function codeberg_browse(lk)
+  local builder = ""
```

Comment:
hi @bagnaram, can you try to implement this API via the `Builder` class? it could help reduce many duplicate lines.


also please update default `router` option here: https://github.com/linrongbin16/gitlinker.nvim/blob/master/lua/gitlinker.lua#L31 using the string template.

and update readme.md:
1. mention `codeberg` is supported: https://github.com/linrongbin16/gitlinker.nvim/blob/master/README.md?plain=1#L22
2. add new router in default option in readme here: https://github.com/linrongbin16/gitlinker.nvim/blob/master/README.md?plain=1#L177
3. mention new API here: https://github.com/linrongbin16/gitlinker.nvim/blob/master/README.md?plain=1#L281

and I didn't see the `blame` router, would you please help with that?

Reply:
for how to use the `Builder`, please check the implementation of `github_browse`: https://github.com/linrongbin16/gitlinker.nvim/blob/master/lua/gitlinker/routers.lua#L94.

Reply:
Thank you for the suggestions on this initial draft of the Codeberg router. I will make the changes mentioned